offline.js

```diff
@@ -262,7 +267,13 @@ const loadOfflineSection = async function(ref, versions, fallbackOnDefaultVersio
 
     const [metadata, fileNameStem] = await loadOfflineSectionMetadataWithCache(ref);
     const textByLang = await loadOfflineSectionByVersions(versions, metadata.versions, metadata.sectionRef, fileNameStem, fallbackOnDefaultVersions);
-    return createFullSectionObject(metadata, textByLang);
+    const fullSectionObject = createFullSectionObject(metadata, textByLang);
+    for (const lang in versions) {
```

Comment:
i'm not sure about that. this uses `versions` which is not passed into `createFullSectionObject`. we can pass it, but i think that it's missing there because it's logically relevat. this function reform the raw data. here we want to handle missing data which maybe belongs more to `loadOfflineSection`.
maybe we should add `addMissingsToFullSectionObject` function? (can we shorten this name?)
(i'm also not sure it's a bad idea, i'm just not sure. lmk what you think)

Reply:
You only need the languages in `versions` and I think these might already exist in `textByLang`. It does full like it's part of creating the full object because it can be derived from `textByLang`.

Reply:
in `textByLang` we'll have an empty array for this language, but this also happens when there is no 'en' at all. in this case it's not missing and we don't want to request it by api. that's the reason i've use `versions` for this.

Reply:
Ok, even if it's no in `textByLang` we can pass a parameter like `requestedLangs` to `createFullSectionObject` which is derived from versions. Unrelated but I just noticed you wrote `for (const lang in versions)`. I think it's more correct to do `for (const lang of versions)`. It's slightly different in JS.

Reply:
for of is for arrays, but i should have use Object.keys